src/client/dfuse/il/int_posix.c

```diff
@@ -88,15 +110,21 @@ ioil_shrink(struct ioil_cont *cont)
 
 	if (cont->ioc_dfs != NULL) {
 		rc = dfs_umount(cont->ioc_dfs);
-		if (rc != 0)
+		if (rc != 0) {
 			D_ERROR("dfs_umount() failed, %d\n", rc);
+			return;
```

Comment:
If the umount fails then the cont_close would also fail.  There's a weird thing going on with ioil_shrink however because it's called once on close, but also once again when the program is exiting so if we inject an error into dfs_umount() then it'll fail the first time, but the second time will work and daos_cont_close will as well.

The code as it was before would fail dfs_umount, then go into daos_cont_close() and through a bunch of needless errors.